### PR TITLE
solution

### PR DESCRIPTION
**Pull request by serentinos**

### CODE REVIEW

**Comment by serentinos** [comment]
Invalid checks because of linter
image

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -1,15 +1,53 @@
...
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  static newAnimal(animalObj) {
+    this.alive.push(animalObj);
+  }
+
+  static animalDeath(animalObj) {
```
Comment (naziamoff) [inline src/herbivoresAndCarnivores.js:10]: is this a name for method, or for the variable?

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  static newAnimal(animalObj) {
```
Comment (naziamoff) [inline src/herbivoresAndCarnivores.js:6]: is this a name for method, or for the variable?

File: src/herbivoresAndCarnivores.js (hunk 1 above)
Comment (naziamoff) [inline src/herbivoresAndCarnivores.js:10]: don't use data type in the variable name

File: src/herbivoresAndCarnivores.js [hunk 3]
```diff
@@ -1,15 +1,53 @@
...
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
```
Comment (naziamoff) [inline src/herbivoresAndCarnivores.js:33]: is `hide` method supposed to be toggled, or it is meant just for hiding?

File: src/herbivoresAndCarnivores.js [hunk 4]
```diff
@@ -1,15 +1,53 @@
...
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    const bitePower = 50;
+    const isHerbivore = Object.hasOwnProperty.call(victim, 'hidden');
```
Comment (naziamoff) [inline src/herbivoresAndCarnivores.js:41]: there's a special method for checking what instance our object is. Checking for some exact property is a crutch.
What if we add the `hidden` property to the Carnivores?

**Review by naziamoff — CHANGES_REQUESTED** [review]

File: src/herbivoresAndCarnivores.js [hunk 5]
```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static addToAlive(animalObj) {
+    this.alive.push(animalObj);
+  }
+
+  static removeFromAlive(animalObj) {
```
Comment (naziamoff) [inline src/herbivoresAndCarnivores.js:8]: - let's use more descriptive names for your variables. It should be maximally clear what your method does by its name
```suggestion
  static removeDeadAnimal(animalObj) {
```

Comment (naziamoff) [reply src/herbivoresAndCarnivores.js:8]: also, you should not use data type in the name of your variables, as it does not give us any additional information. It should have clear naming even without explicit mention of data type

File: src/herbivoresAndCarnivores.js [hunk 6]
```diff
@@ -1,15 +1,53 @@
...
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    const bitePower = 50;
+    const isHerbivore = victim instanceof Herbivore;
+
+    if (!victim.hidden && isHerbivore) {
```
Comment (naziamoff) [inline src/herbivoresAndCarnivores.js:43]: better to switch the places. If Carnivore is passed you have redundant check for `hidden`

File: src/herbivoresAndCarnivores.js (hunk 5 above)
Comment (serentinos) [reply src/herbivoresAndCarnivores.js:8]: Thank you for suggestion!

**Review by 33Orange — APPROVED** [review]
Well done !

**Review by naziamoff — APPROVED** [review]

**Pull request closed without merging**